ThunderCloud/StormLanguageController.swift

```diff
@@ -217,45 +216,71 @@ public class StormLanguageController: NSObject {
         return (regionalLanguagePack: regionalLanguagePack, majorLanguagePack: majorLanguagePack)
     }
     
-    /// Reloads the language pack based on user preferences and assigns it to the language dictionary
-    public func reloadLanguagePack() {
+    /// Loads the language dictionary from the specified path, if it exists, and writes it into the provided dictionary.
+    ///
+    /// - Parameters:
+    ///   - path: The path that the language data is stored at.
+    ///   - dictionary: The dictionary that the language data should be read in to.
+    private func writeLanguageData(from path: String, into dictionary: inout [AnyHashable: Any]) {
+        guard let languageData = languageDictionary(for: path) else {
+            return
+        }
         
-        //Load languages
+        for (key, value) in languageData {
+            dictionary[key] = value as Any
+        }
+    }
+    
+    /// Reloads the language pack based on override language requirements, or device language and region.
+    ///
+    /// We'll firstly attempt to read from an overriden language pack (if one is set), if we can't to that then we'll see what the device has picked up and use that language pack instead.
+    public func reloadLanguagePack() {
         var finalLanguage = [AnyHashable: Any]()
         
+        // If we have an override set to use, lets try to use that instead of detecting language packs here.
+        // If we can't use it, then we'll fallback to use normal detection.
+        if let overrideLanguagePack = overrideLanguagePack {
+            let overrideFileName = overrideLanguagePack.fileName
+            
+            currentLanguage = overrideFileName
+            
+            // Handle the major language element of this language (i.e., eng from usa_eng). If there is no region specified, this will just return the language.
+            if let majorLanguage = overrideFileName.components(separatedBy: "_").last, let majorPackPath = ContentController.shared.fileUrl(forResource: majorLanguage, withExtension: "json", inDirectory: "languages") {
+                writeLanguageData(from: majorPackPath.path, into: &finalLanguage)
+                
+                // Now attempt to handle the regional version of the language (i.e., usa_eng). If there was no region specified and we've already handled this above, we skip this as it's unnecessary.
+                if let regionalLanguagePath = ContentController.shared.fileUrl(forResource: overrideFileName, withExtension: "json", inDirectory: "languages"), majorLanguage != overrideFileName {
```

Comment:
The `majorLanguage != overrideFileName` detects here if I'm correct that the currentLanguage is just a base language, for example `en` rather than `en_us` correct? Maybe put that in the second sentence of the comment as took me a little while to work out why that was there! 😊

Reply:
Will make a note of that in the comment. That is indeed correct :)

Reply:
@simonmitchell It seems the documentation was there, but as it was inline with the first sentence it seemed quite unclear. Shifted it around, and renamed a few vars in 32d708b20079acc3487949b863ba462245b06852 :)